### PR TITLE
Refactor EntitySelectionTree

### PR DESCRIPTION
**Pull request by AnthonyMDev**
This was a major refactoring of the EntitySelectionTree. It's slimmer and easier to reason about and debug now.

There were a few bugs resolved by this PR, specifically surrounding computation of merged fields when merging named fragment spreads with inclusion conditions.

### CODE REVIEW

File: Sources/StarWarsAPI/StarWarsAPI/Sources/Fragments/CharacterNameWithInlineFragment.graphql.swift [hunk 1]
```diff
@@ -63,8 +63,6 @@ public struct CharacterNameWithInlineFragment: StarWarsAPI.SelectionSet, Fragmen
 
       /// The movies this character appears in
       public var appearsIn: [GraphQLEnum<StarWarsAPI.Episode>?] { __data["appearsIn"] }
-      /// The name of the character
```
Comment (calvincestari) [inline Sources/StarWarsAPI/StarWarsAPI/Sources/Fragments/CharacterNameWithInlineFragment.graphql.swift:66]: This is an odd field to be removing; is it just catching up on generated code changes from prior PRs or is it relevant to this PR?

**Review by calvincestari — APPROVED** [review]
There's a lot going on in this PR, thanks for the run-through earlier; I focused my review on the generated code changes and new edge case tests. Thanks for finding and fixing these bugs.
